src/Omega_h_meshsim.cpp

```diff
@@ -178,40 +342,41 @@ void read_internal(pMesh m, Mesh* mesh) {
     }
   }
   auto quad2verts = Read<LO>(host_quad2verts.write());
-  down = reflect_down(quad2verts, edge2vert.ab2b, vert2edge, Topo_type::quadrilateral, Topo_type::edge);
-  mesh->set_ents(Topo_type::quadrilateral, Topo_type::edge, down);
-
-  RIter regions = M_regionIter(m);
-  LO count_tet = 0;
-  LO count_hex = 0;
-  LO count_wedge = 0;
-  LO count_pyramid = 0;
-  pRegion rgn;
-  while (rgn = (pRegion) RIter_next(regions)) {
-    if (R_topoType(rgn) == Rtet) {
-      count_tet += 1;
-    }
-    else if (R_topoType(rgn) == Rhex) {
-      count_hex += 1;
-    }
-    else if (R_topoType(rgn) == Rwedge) {
-      count_wedge += 1;
-    }
-    else if (R_topoType(rgn) == Rpyramid) {
-      count_pyramid += 1;
-    }
-    else {
-      Omega_h_fail("Region is not tet, hex, wedge, or pyramid \n");
-    }
+  if (is_hypercube) {
+    down = reflect_down(quad2verts, edge2vert.ab2b, vert2edge,
+                        OMEGA_H_HYPERCUBE, 2, 1);
+    mesh->set_ents(2, down);
+    mesh->add_tag<ClassId>(2, "class_id", 1,
+                           Read<ClassId>(host_class_ids_face.write()));
+    mesh->add_tag<I8>(2, "class_dim", 1,
+                      Read<I8>(host_class_dim_face.write()));
+  }
+  else if (is_simplex) {
```

Comment:
We should add a comment for future readers; the empty conditional is a bit puzzling at first sight.